### PR TITLE
Update docs per rebar3_hex v7 release

### PR DESCRIPTION
**Pull request by starbelly**
This is an update to the docs per v7.0.0. I think we could better flesh these out, but I'd like to get the bare minimum changes out to start with.

### CODE REVIEW

**Review by ferd — APPROVED** [review]

File: lib/hexpm_web/templates/docs/rebar3_publish.html.md [hunk 1]
```diff
@@ -57,8 +57,15 @@ You can also add any of the following to the list of application attributes:
   <dd>A map where the key is a link name and the value is the link URL. Optional but highly recommended.</dd>
   <dt><code>files</code></dt>
   <dd>A list of files and directories to include in the package. Defaults to standard project directories, so you usually don't need to set this property.</dd>
+  <dt><code>include_paths</code></dt>
+  <dd>A list of paths containing files you wish to include in a release.</dd>
+  <dt><code>exclude_paths</code></dt>
+  <dd>A list of paths containing files you wish to exclude in a release.</dd>
```
Comment (ericmj) [inline lib/hexpm_web/templates/docs/rebar3_publish.html.md:63]:
```suggestion
  <dt><code>include_paths</code></dt>
  <dd>A list of paths containing files you wish to include in the package.</dd>
  <dt><code>exclude_paths</code></dt>
  <dd>A list of paths containing files you wish to exclude in the package.</dd>
```

File: lib/hexpm_web/templates/docs/rebar3_publish.html.md [hunk 2]
```diff
@@ -57,8 +57,15 @@ You can also add any of the following to the list of application attributes:
...
+  <dt><code>exclude_patterns</code></dt>
+  <dd>A list of regular expressions used to exclude files that may have been accumulated via files and include_paths and standard project paths.</dd>
   <dt><code>build_tools</code></dt>
   <dd>List of build tools that can build the package. It's very rare that you need to set this.</dd>
+
```
Comment (ericmj) [inline lib/hexpm_web/templates/docs/rebar3_publish.html.md:68]:
```suggestion
```

File: lib/hexpm_web/templates/docs/rebar3_publish.html.md [hunk 3]
```diff
@@ -101,6 +108,29 @@ Only Hex packages may be used as dependencies of the package. It is not possible
 
 ```
 
+#### Documentation 
+
+rebar3_hex by default expects you to configure a documentation provider. We recommend using 
+[rebar3_ex_doc](https://hexdocs.pm/rebar3_ex_doc/) for publishing documentation along with your package for a consistent 
+format and style on [hex.pm](https://hex.pm/). 
+
+
```
Comment (ericmj) [inline lib/hexpm_web/templates/docs/rebar3_publish.html.md:117]:
```suggestion
```

File: lib/hexpm_web/templates/docs/rebar3_publish.html.md [hunk 4]
```diff
@@ -101,6 +108,29 @@ Only Hex packages may be used as dependencies of the package. It is not possible
...
+```erlang 
+ex_doc, [
```
Comment (ericmj) [inline lib/hexpm_web/templates/docs/rebar3_publish.html.md:119]:
```suggestion
{ex_doc, [
```

File: lib/hexpm_web/templates/docs/rebar3_publish.html.md [hunk 5]
```diff
@@ -101,6 +108,29 @@ Only Hex packages may be used as dependencies of the package. It is not possible
...
+    {source_url, <<"https://github.com/namespace/your_app">>},
+    {extras, [<<"README.md">>, <<"LICENSE">>]},
+    {main, <<"readme">>}
+]}.
+
+{hex, [{doc, ex_doc}]}.
+```
+
+Alternatively, or on Erlang versions older than OTP-24, you can use the edoc provider that ships with rebar3 :
```
Comment (ericmj) [inline lib/hexpm_web/templates/docs/rebar3_publish.html.md:128]:
```suggestion
Alternatively, or on Erlang versions older than OTP-24, you can use the edoc provider that ships with rebar3: 
```

**Review by ericmj — APPROVED** [review]

**Comment by ericmj** [comment]
Thank you @starbelly!

**Pull request merged**